### PR TITLE
TSWEB-957 tests for renewable energy page

### PR DESCRIPTION
**Pull request by Valerii-Kryvokhyzha**

### CODE REVIEW

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	const testData =
+		'https://www2.deloitte.com/content/dam/insights/us/articles/6387_100-Percent-Renewables/DI_100-Percent-Renewables.pdf';
+
+	const actualLink = await buttonDeloitte.getAttribute('href');
+
+	expect(actualLink).toBe(testData);
+
+	await buttonDeloitte.click();
+	const newPage = await driver.DriverContext.waitForEvent('page');
+	await newPage.waitForLoadState('networkidle');
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyActions.test.ts:37]: where is the check?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyActions.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+test("Check carousel sections, arrows and CTA button in 'How We Operate at Techstack' container from the 'Renewable Energy' block @Regression @RenewableEnergy @TSWEB-957", async () => {
+	const howWeOperateContainer = driver.getByTestId(RenewableEnergy.HowWeOperateAtTechstack);
+	const carousel = howWeOperateContainer.getByTestId(Container.ContainerCarousel);
+
+	const allSectionTitles = await carousel.getByTestId(Container.SectionTitle).allInnerTexts();
+	const testData = ['Make\ncontact', 'Speak with\na tech expert', 'Making\na proposal', 'Contract\nsigning'];
+
+	expect(allSectionTitles.sort()).toEqual(testData.sort());
+	expect(await carousel.getByTestId(Container.SectionNumber).allInnerTexts()).toEqual([
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyActions.test.ts:100]: Can you check `await expect(carousel.getByTestId(.......`

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	await expect(driver.getByTestId(RenewableEnergy.Faq).getByTestId(Container.ContainerTitle)).toHaveText('FAQ');
+	await expect(driver.getByTestId(RenewableEnergy.Faq).getByTestId(Container.ContainerNumber)).toHaveText('09');
+});
+
+test("Check block titles in 'Techstack in Numbers' container from the 'Renewable Energy' block @Regression @RenewableEnergy @TSWEB-957", async () => {
+	const techstackInNumbersContainer = driver.getByTestId(RenewableEnergy.TechstackInNumbers);
+	const allBlockTitles = await techstackInNumbersContainer.getByTestId(Container.BlockTitle).allInnerTexts();
+	const testData = ['9\nyears', '20\nexperts', '91\n%', '2 of 3\nclients'];
+
+	expect(allBlockTitles.sort()).toEqual(testData.sort());
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts:90]: why did you use sort()?

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+});
+
+test("Check section titles and numbers in 'Who We Serve' container from the 'Renewable Energy' block @Regression @RenewableEnergy @TSWEB-957", async () => {
+	const whoWeServeContainer = driver.getByTestId(RenewableEnergy.WhoWeServe);
+
+	expect(await whoWeServeContainer.getByTestId(Container.SectionNumber).allInnerTexts()).toEqual([
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts:96]: same

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+test("Check block titles, section titles and CTA button in 'Renewable Energy Software Development Services' container from the 'Renewable Energy' block @Regression @RenewableEnergy @TSWEB-957", async () => {
+	const renewableEnergySoftDevServContainer = driver.getByTestId(RenewableEnergy.RenewableEnergySoftDevServ);
+	const allBlockTitles = await renewableEnergySoftDevServContainer.getByTestId(Container.BlockTitle).allInnerTexts();
+	const testDataBlockTitles = [
+		'Domain-Specific Software for the Renewable Energy Industry',
+		'Energy Management Systems',
+		'Energy Infrastructure\nand Analysis',
+	];
+	expect(allBlockTitles.sort()).toEqual(testDataBlockTitles.sort());
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts:122]: same

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+	}
+
+	await expect(renewableEnergySoftDevServContainer.getByTestId(MainSiteButtons.BookAMeeting)).toBeVisible();
+});
+
+test("Check section titles, block title, image and CTA button in 'The Solar Energy Data Portal by Techstack' container from the 'Renewable Energy' block @Regression @RenewableEnergy @TSWEB-957", async () => {
+	const theSolarEnergyContainer = driver.getByTestId(RenewableEnergy.TheSolarEnergyDataPortalByTechstack);
+	const allSectionTitles = await theSolarEnergyContainer.getByTestId(Container.SectionTitle).allInnerTexts();
+	const testData = ['Improved efficiency', 'Energy-centric system for customers', 'Data normalization'];
+	expect(allSectionTitles.sort()).toEqual(testData.sort());
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts:139]: same

File: ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+	const blockTitle = await theSolarEnergyContainer.getByTestId(Container.BlockTitle).innerText();
+	expect(blockTitle).toEqual('Solar energy data portal scheme');
```
Comment (anastasiia-liashova) [inline ui_tests_playwright/automation/test/ui/MainSite/industries/RenewableEnergy/RenewableEnergyContent.test.ts:142]: await expect ().haveText()

**Review by anastasiia-liashova — CHANGES_REQUESTED** [review]

**Review by anastasiia-liashova — APPROVED** [review]

**Review by DaniilVostrikov — APPROVED** [review]

**Pull request merged**